tokens/usecase/tokens_usecase.go

```diff
@@ -349,40 +390,31 @@ func (t *tokensUseCase) getPricesForBaseDenom(ctx context.Context, baseDenom str
 }
 
 func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, bool) {
-	result, ok := t.precisionScalingFactorMap[precision]
-	return result, ok
-}
-
-// GetTokensFromChainRegistry fetches the tokens from the chain registry.
-// It returns a map of tokens by chain denom.
-func GetTokensFromChainRegistry(chainRegistryAssetsFileURL string) (map[string]domain.Token, error) {
-	// Fetch the JSON data from the URL
-	response, err := http.Get(chainRegistryAssetsFileURL)
-	if err != nil {
-		return nil, err
+	result, ok := t.precisionScalingFactorMap.Load(precision)
+	if !ok {
+		return osmomath.Dec{}, false
 	}
-	defer response.Body.Close()
 
-	// Decode the JSON data
-	var assetList AssetList
-	err = json.NewDecoder(response.Body).Decode(&assetList)
-	if err != nil {
-		return nil, err
+	v, ok := result.(osmomath.Dec)
+	if !ok {
+		return osmomath.Dec{}, false
 	}
 
-	tokensByChainDenom := make(map[string]domain.Token)
+	return v, true
+}
 
-	// Iterate through each asset and its denom units to print exponents
-	for _, asset := range assetList.Assets {
-		token := domain.Token{}
-		token.Precision = asset.Decimals
-		token.HumanDenom = asset.Symbol
-		token.IsUnlisted = asset.Preview
-		token.CoingeckoID = asset.CoingeckoID
-		tokensByChainDenom[asset.CoinMinimalDenom] = token
+// UpdateAssetsAtHeightIntervalSync updates assets at configured height interval.
+func (t *tokensUseCase) UpdateAssetsAtHeightIntervalSync(height uint64) error {
+	if t.tokenLoader == nil {
+		return nil // no-op, no token loader is set
```

Comment:
IMO we can remove this check

If the app wiring issue occurs, I would rather have it panic early due to nil pointer dereferencing rather than battling a prod issue debugging why the asset list doesn't update and there is no error

Reply:
Updated as per your suggestion.